src/lime/tools/XCodeHelper.hx

```diff
@@ -173,22 +207,32 @@ class XCodeHelper
 
 	public static function getSimulatorID(project:HXProject):String
 	{
-		var simulator = getSelectedSimulator(project);
-		if (simulator == null)
-		{
-			return null;
-		}
-		return simulator.id;
+		var simulator = getPreferredSimulator(project);
+		return (simulator != null) ? simulator.id : null;
 	}
 
 	public static function getSimulatorName(project:HXProject):String
 	{
-		var simulator = getSelectedSimulator(project);
-		if (simulator == null)
+		var simulator = getPreferredSimulator(project);
+		return (simulator != null) ? simulator.name : null;
+	}
+
+	private static function getPreferredSimulator(project:HXProject):SimulatorInfo
+	{
+		// try and get the simulator which is currently open and running
+		// if there are none running then fall back to getting the selected simulator
+		var simulator = getBootedSimulator();
+		if (simulator != null) 
+		{
+			return simulator;
+		}
+
+		simulator = getSelectedSimulator(project);
+		if (simulator != null)
 		{
-			return null;
+			return simulator;
 		}
-		return simulator.name;
+		return null;
```

Comment:
`if (simulator != null) return simulator else return null` can be simplified to `return simulator`.